src/games/strategy/triplea/ai/proAI/util/ProBattleUtils.java

```diff
@@ -207,17 +215,17 @@ public ProBattleResult callBattleCalculator(final PlayerID player, final Territo
     final int runCount = Math.max(16, 100 - minArmySize);
     if (isAttacker) {
       if (retreatWhenOnlyAirLeft) {
-        ai.getCalc().setRetreatWhenOnlyAirLeft(true);
+        calc.setRetreatWhenOnlyAirLeft(true);
```

Comment:
boolean parameter code smell, can be fixed by refactoring the public API to have two methods, one of them woudl be:  `calc.retreatWhenOnlyAirLeft()`